controllers/nutanixcluster_controller.go

```diff
@@ -289,20 +328,81 @@ func (r *NutanixClusterReconciler) reconcileNormal(rctx *nctx.ClusterContext) (r
 
 func (r *NutanixClusterReconciler) reconcileFailureDomains(rctx *nctx.ClusterContext) error {
 	log := ctrl.LoggerFrom(rctx.Context)
-	if len(rctx.NutanixCluster.Spec.FailureDomains) == 0 {
-		log.V(1).Info("no failure domains defined on cluster")
-		conditions.MarkTrue(rctx.NutanixCluster, infrav1.NoFailureDomainsReconciled)
+	log.Info("Reconciling failure domains for cluster")
+
+	failureDomains := capiv1.FailureDomains{}
+	if len(rctx.NutanixCluster.Spec.ControlPlaneFailureDomains) == 0 && len(rctx.NutanixCluster.Spec.FailureDomains) == 0 { //nolint:staticcheck // suppress complaining on Deprecated field
+		log.Info("No failure domains configured for cluster.")
+		conditions.MarkTrue(rctx.NutanixCluster, infrav1.NoFailureDomainsConfiguredCondition)
+		conditions.Delete(rctx.NutanixCluster, infrav1.FailureDomainsValidatedCondition)
+
+		// Reset the failure domains for nutanixcluster status
+		rctx.NutanixCluster.Status.FailureDomains = failureDomains
 		return nil
 	}
-	log.V(1).Info("Reconciling failure domains for cluster")
-	// If failure domains is nil on status object, first create empty slice
-	if rctx.NutanixCluster.Status.FailureDomains == nil {
-		rctx.NutanixCluster.Status.FailureDomains = make(capiv1.FailureDomains, 0)
+
+	// Clear NoFailureDomainsConfiguredCondition condition
+	conditions.Delete(rctx.NutanixCluster, infrav1.NoFailureDomainsConfiguredCondition)
+
+	validationErrs := []error{}
+	for _, fdRef := range rctx.NutanixCluster.Spec.ControlPlaneFailureDomains {
+		// Fetch the referent failure domain object
+		fdObj := &infrav1.NutanixFailureDomain{}
+		fdKey := client.ObjectKey{Name: fdRef.Name, Namespace: rctx.NutanixCluster.Namespace}
+		err := r.Client.Get(rctx.Context, fdKey, fdObj)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				validationErrs = append(validationErrs, fmt.Errorf("not found the failure domain object with name %q", fdRef.Name))
+			} else {
+				validationErrs = append(validationErrs, fmt.Errorf("failed to fetch the failure domain object with name %q: %w", fdRef.Name, err))
+			}
+			continue
```

Comment:
you can put this continue in

```
if errors.IsNotFound(err) {
				validationErrs = append(validationErrs, fmt.Errorf("not found the failure domain object with name %q", fdRef.Name))
				continue
			}
			
```

and remove the else

Reply:
actually never mind. i was wrong on this

Reply:
changed as suggested